### ISSUE TITLE
DataFrameMapper fit doesn't take y argument

### ISSUE THREAD

**Issue by vzaretsk** [paulgb/sklearn-pandas#58]
Currently DataFrameMapper fit method doesn't take a y argument. I have a use case that needs this (I'm doing supervised dimensionality reduction) and made a small modification to enable this functionality. If there is interest, I can submit a pull request with these changes. Additionally, it seems that this would eliminate the need for a custom Pipeline class.

**Comment by dukebody** [comment]
Hi @vzaretsk . There has been some discussion regarding this feature already: https://github.com/paulgb/sklearn-pandas/pull/54

I feel I may be behaving a bit too defensive here. I believe this could be a good feature to have as long as:

- Existing code works unchanged. Deprecation plans are acceptable.
- The `DataFramaMapper` class doesn't get too complex with a lot larger surface for bugs.

Please submit the PR together with a use case of the change, so we can find out the best way to implement this y-transforming feature.

**Comment by vzaretsk** [comment]
My use case is for the shelter animal outcomes Kaggle competition. Rather than creating over 100 dummy variables for the various cat and dog breeds, I replaced each breed by the average outcome per breed. The y (outcome) information is used during fit to find this average outcome for each breed in the CV fold. Other use cases could be supervised clustering or using LDA to find the direction that best separates the classes.
I submitted the PR. All tests are passing although I will need to make some additional ones if this goes forward. Existing code should work as is.
